local_planner/src/nodes/common.h

```diff
@@ -10,58 +10,53 @@
 
 namespace avoidance {
 
-float distance2DPolar(int e1, int z1, int e2, int z2);
+struct PolarPoint {
+  float e;
+  float z;
+  float r;
+};
+
+float distance2DPolar(const PolarPoint& p1, const PolarPoint& p2);
 
 /**
 * @brief     Convertes the point from polar CS to cartesian CS
-* @param[in] e elevation angle (-90, 90]
-* @param[in] z azimuth angle (-180,180]
-* @param[in] radius
+* @param[in] p_pol polar point
 * @param[in] pos Position from which to convert the point
 * @returns   point in cartesian CS
 **/
-Eigen::Vector3f fromPolarToCartesian(float e, float z, double radius,
-                                     const geometry_msgs::Point& pos);
+Eigen::Vector3f PolarToCartesian(const PolarPoint& p_pol,
+                                 const geometry_msgs::Point& pos);
 double indexAngleDifference(float a, float b);
 
-double elevationIndexToAngle(int e, double res);
-double azimuthIndexToAngle(int z, double res);
+PolarPoint HistogramIndexToPolar(int e, int z, int res, const float& radius);
 
 /**
-* @brief     Compute the bearing angle of a given vector relative to an
+* @brief     Compute the polar point to an
```

Comment:
Careful... The brief is no longer really explaining much. Compute the polar point to an origin? What?